sensor/kubernetes/listener/listener_impl.go

```diff
@@ -29,9 +32,22 @@ type listenerImpl struct {
 	traceWriter        io.Writer
 	outputQueue        component.Resolver
 	storeProvider      *resources.InMemoryStoreProvider
+	context            context.Context
+	contextMtx         sync.Mutex
+}
+
+func (k *listenerImpl) StartWithContext(ctx context.Context) error {
+	k.contextMtx.Lock()
+	defer k.contextMtx.Unlock()
+	k.context = ctx
+	return k.Start()
 }
 
 func (k *listenerImpl) Start() error {
+	if k.context == nil {
+		return errors.New("cannot start listener without a context")
```

Comment:
Should we maybe panic here in dev mode (while on release still returning an error)?